src/System.Net.Primitives/tests/FunctionalTests/IPAddressTest.cs

```diff
@@ -242,23 +247,55 @@ public static void Equals_Compare_Success()
             Assert.False(ip4.GetHashCode().Equals(ip6.GetHashCode()));
         }
 
+        [Theory]
+        [MemberData(nameof(GetValidIPAddresses))]
+        public static void GetHashCode_ValidIPAddresses_Success(IPAddress ipAddress)
+        {
+            ipAddress.GetHashCode();
+        }
+
+        private static IEnumerable<object[]> GetValidIPAddresses()
+        {
+            return IPAddressParsing.ValidIpv4Addresses
+                .Concat(IPAddressParsing.ValidIpv6Addresses)
+                .Select(array => new object[] {IPAddress.Parse((string)array[0])});
+        }
+
+        [Theory]
+        [MemberData(nameof(GeneratedIPAddresses))]
+        public static void GetHashCode_VariousIPAddressGenerators_Success(IPAddress ipAddress)
+        {
+            ipAddress.GetHashCode();
```

Comment:
Ditto; I think it makes sense as Dan suggested to change this to something like:
```C#
Assert.Equal(ipAddress.GetHashCode(), ipAddress.GetHashCode());
```